File: topic/knowledge-qa/jxin/backend/app/application/dto/document.py
# rag_demo/backend/app/application/dto/document.py
from pydantic import BaseModel
from datetime import datetime
from typing import List


class UploadResponse(BaseModel):
    """上传响应"""
    doc_id: str
    filename: str
    chunks: int


class DocumentDTO(BaseModel):
    """文档 DTO"""
    id: str
    filename: str
    chunk_count: int
    created_at: datetime


class DocumentListResponse(BaseModel):
    """文档列表响应"""
    documents: List[DocumentDTO]


class DeleteResponse(BaseModel):
    """删除响应"""
    success: bool
    message: str = ""

File: topic/knowledge-qa/jxin/frontend/src/api/document.ts
// rag_demo/frontend/src/api/document.ts
import request from './request'

export interface UploadResponse {
  doc_id: string
  filename: string
  chunks: number
}

export interface Document {
  id: string
  filename: string
  chunk_count: number
  created_at: string
}

export interface DocumentListResponse {
  documents: Document[]
}

export interface DeleteResponse {
  success: boolean
  message: string
}

export const documentApi = {
  upload: (file: File): Promise<UploadResponse> => {
    const formData = new FormData()
    formData.append('file', file)
    return request.post('/documents/upload', formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    })
  },

  list: (): Promise<DocumentListResponse> =>
    request.get('/documents'),

  delete: (docId: string): Promise<DeleteResponse> =>
    request.delete(`/documents/${docId}`),
}
